### PR TITLE
Do strict DNS lookup on ACME

### PR DESCRIPTION
**Pull request by maraino**
This commit changes the ACME challenges to perform a strict DNS lookup without taking into account the search list in a resolv.conf

### CODE REVIEW

**Review by dopey — APPROVED** [review]

File: acme/challenge_test.go [hunk 1]
```diff
@@ -457,7 +457,7 @@ func TestChallenge_Validate(t *testing.T) {
 						assert.Equal(t, "zap.internal", updch.Value)
 						assert.Equal(t, StatusPending, updch.Status)
 
-						err := NewError(ErrorConnectionType, "error doing http GET for url http://zap.internal/.well-known/acme-challenge/%s: force", ch.Token)
+						err := NewError(ErrorConnectionType, "error doing http GET for url http://zap.internal./.well-known/acme-challenge/%s: force", ch.Token)
```
Comment (hslatman) [inline acme/challenge_test.go:460]: I think we shouldn't report the dot in the error messages, as it's different from what the user requested (most likely). Is it possible to keep the rooted name strictly internally?

File: test/integration/scep/common_test.go [hunk 2]
```diff
@@ -39,6 +39,9 @@ func newCAClient(t *testing.T, caURL, rootFilepath string) *ca.Client {
 }
 
 func requireHealthyCA(t *testing.T, caClient *ca.Client) {
+	// Wait for CA
+	time.Sleep(time.Second)
```
Comment (hslatman) [inline test/integration/scep/common_test.go:43]: I see I forgot this before:
```suggestion
func requireHealthyCA(t *testing.T, caClient *ca.Client) {
        t.Helper()
	// Wait for CA
	time.Sleep(time.Second)
```

File: acme/challenge_test.go (hunk 1 above)
Comment (dopey) [reply acme/challenge_test.go:460]: Agreed. Maybe could be an internal field on the Challenge?

Comment (maraino) [reply acme/challenge_test.go]: Fixed with 2b30ae50

**Review by dopey — APPROVED** [review]

**Review by hslatman — APPROVED** [review]
Great 😄

**Pull request merged**